### PR TITLE
Do not disable compaction on sidecar without object storage configuration

### PR DESCRIPTION
**Pull request by bwplotka**
... as it's not required (:

Happy Halloween! :jack_o_lantern: 

Also:
* updated Thanos docs!
* added tests
* bumped default image.

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/prometheus/statefulset.go [hunk 1]
```diff
@@ -816,9 +819,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		if p.Spec.LogFormat != "" {
 			container.Args = append(container.Args, fmt.Sprintf("--log.format=%s", p.Spec.LogFormat))
 		}
-
 		additionalContainers = append(additionalContainers, container)
-		promArgs = append(promArgs, "--storage.tsdb.min-block-duration=2h", "--storage.tsdb.max-block-duration=2h")
```
Comment (paulfantom) [inline pkg/prometheus/statefulset.go]: This will remove tsdb compaction settings even when `p.Spec.Thanos.ObjectStorageConfig == nil`.  Do we want that?

Comment (paulfantom) [reply pkg/prometheus/statefulset.go]: nvm, I need coffee :smile:

File: Documentation/thanos.md [hunk 2]
```diff
@@ -2,32 +2,55 @@
...
+Prometheus Operator operates Prometheus, not Thanos. However Thanos system integrates with existing setup by adding 
+sidecar to each Prometheus running in the system.
+ 
+Please before continuing with Prometheus Operator Thanos integration, read more about Thanos in the [documentation](https://thanos.io/getting-started.md/).
 
-Now please take the time to look at the Thanos README and read through the documentation before continuing with the Prometheus Operator integration.
+Prometheus Operator allows your to optionally add Thanos sidecar to Prometheus. Sidecar allows to hook into Thanos
+querying system as well as **optionally** back up your data in object storage.
+ 
+Thanos system includes other components like the queries or rulers. To get advantage of object storage it also requires compactors, and stores.
```
Comment (paulfantom) [inline Documentation/thanos.md]:
```suggestion
Thanos system includes other components like queriers or rulers. To get the advantage of object storage it also requires compactors and stores.
```

File: Documentation/thanos.md [hunk 3]
```diff
@@ -2,32 +2,55 @@
...
+...
+```
+Note: If you're using Istio you may need to also set `ListenLocal` on the Thanos spec due to Istio's forwarding of traffic to localhost.
+
+### Optional: Configuring Thanos Object Storage
+
+If you want sidecar to be able to upload blocks to object storage you need to tell Prometheus Operator about it.
 
-Beginning with Thanos v0.2 the sidecar assumes an existing Kubernetes Secret containing the Thanos configuration.
+In this mode sidecar assumes athexisting Kubernetes Secret containing the Thanos configuration.
```
Comment (brancz) [inline Documentation/thanos.md]: athexisting?

Comment (paulfantom) [inline Documentation/thanos.md]:
```suggestion
In this mode, sidecar assumes an existing Kubernetes Secret containing the Thanos configuration.
```

File: pkg/prometheus/statefulset.go [hunk 4]
```diff
@@ -35,7 +35,7 @@ import (
 const (
 	governingServiceName            = "prometheus-operated"
 	DefaultPrometheusVersion        = "v2.7.1"
```
Comment (bwplotka) [inline pkg/prometheus/statefulset.go]: Can I update this?

File: pkg/prometheus/statefulset.go (hunk 1 above)
Comment (bwplotka) [reply pkg/prometheus/statefulset.go]: =D

File: pkg/prometheus/statefulset.go (hunk 4 above)
Comment (brancz) [reply pkg/prometheus/statefulset.go]: please do

File: Documentation/thanos.md (hunk 3 above)
Comment (s-urbaniak) [reply Documentation/thanos.md]: "an existing"?

**Comment by bwplotka** [comment]
This should be rdy to go - restarted job, looked flaky.

**Comment by brancz** [comment]
I’ve re-triggered this a couple times, it seems to fail consistently.

**Comment by s-urbaniak** [comment]
let me execute that failing test locally on both master and this branch.

**Comment by bwplotka** [comment]
Can repro locally
```
=== CONT  TestAllNS/y/PromArbitraryFSAcc/allowed-tls-file
--- FAIL: TestAllNS (383.70s)
    --- FAIL: TestAllNS/y (0.00s)
        --- FAIL: TestAllNS/y/PromArbitraryFSAcc (0.00s)
            --- FAIL: TestAllNS/y/PromArbitraryFSAcc/allowed-tls-file (325.08s)
                prometheus_test.go:1981: waiting for targets timed out. 0 of 1 targets found. timed out waiting for the condition
FAIL
exit status 1

```

**Comment by bwplotka** [comment]
The reason is the Prometheus version update.

Let's update this in separate PR and figure out the issue there (: Removed from this commit.

**Comment by s-urbaniak** [comment]
LGTM :+1:

**Pull request merged**